core/src/main/scala/org/apache/spark/deploy/worker/ExecutorRunner.scala

```diff
@@ -108,13 +108,12 @@ private[spark] class ExecutorRunner(
   }
 
   /** Replace variables such as {{EXECUTOR_ID}} and {{CORES}} in a command argument passed to us */
-  def substituteVariables(argument: String): String = argument match {
```

Comment:
Of course, I don't mean you would have substitutions for values that don't exist. It would be different between the two. The driver only supports two substitutions now. I suppose I mean I like that you improved the logic to allow substituting these placeholders *within* arguments. That would be nice to have both places.